### ISSUE TITLE
Single quote colour glitch

### ISSUE THREAD

**Issue by onetrev** [dracula/visual-studio-code#60]
As per the screenshot, the single quote and slashes here should not be a different colour from the text. 

![capture](https://user-images.githubusercontent.com/597601/32974771-f7cab852-cc54-11e7-850c-a9a51e6d90a2.PNG)

_LANGUAGE: PHP_

This actually affects a bunch of themes I've seen so far, including about half of the included ones in VS Code. Like Dracula, other themes don't seem to handle this text properly. However some are slightly different in that the entire text in the second single quote is a different colour from the first `if` statement single quote text. So no idea what the issue is.

Seems like it should be controllable since some themes, ie. Cobalt2, doesn't have this. But maybe it's not...?  Thanks!

**Comment by dsifford** [comment]
Thanks for the report.

The issue here is that the grammar for php is non-discriminatory for regular expressions. At any time it sees a string with `/` delimiters, it just assumes you're wanting to do a regex operation and the scope switches entirely.

The reason that Cobalt2 isn't affected is because they likely don't scope regular expressions at all. That's not an ideal solution here because in languages that have deep regular expression grammar baked in (e.g. javascript and typescript), the highlighting of the different groups really comes in handy.

The reason why some other themes are affected, but where the string is entirely a different color than what strings are normally supposed to be is because they likely scope regular expressions, but don't scope it down any further.

This theme has unique highlighting for every scope within a regular expression, which is why you're seeing red delimiters and yellow string-literals. Here's an example of that in typescript:

![image](https://user-images.githubusercontent.com/5240018/32975238-4eb8a6ba-cbd3-11e7-9de4-3b08c8dc96d1.png)

**TL;DR:** The fix would have to be made in the grammar. It cannot be made in the theme. Would you mind opening this issue [here](https://github.com/atom/language-php) instead?

Thanks in advance!

**Comment by onetrev** [comment]
Gotcha. I did figure it might be something out of your control, since it was hit and miss. Sorry.

Happy to post the issue elsewhere so this can be tackled. Only thing is the link you gave me is for Atom -> PHP. Can you confirm where best to post for VS Code PHP issues?

**Comment by dsifford** [comment]
Oh, just realized you also were wondering why the first parsed as a normal string and the second did not..

That's because the first uses the delimiter character inside the "expression", so the regex scope match fails and it defaults back to string.

The second one has the delimiter character only on the outer edges of the string, so the regex scope passes.

the following two strings should also fail regex matching:

```php
<?php

$not_regex_1 = '/some-dirname';
$not_regex_2 = 'some-other-dirname/';
$regex = '/i-am-now-regex/';
```
----

> Gotcha. I did figure it might be something out of your control, since it was hit and miss. Sorry

No apology necessary! I'd prefer things like this to be reported and clarified, rather than not.

> Only thing is the link you gave me is for Atom -> PHP. Can you confirm where best to post for VS Code PHP issues?

Yep, that's the right location. VSCode pulls the PHP grammar directly from Atom.
